sdk/ai/ai-projects/src/agents/utils.ts

```diff
@@ -40,17 +40,154 @@ export enum connectionToolType {
+  if (fileIds && dataSources) {
+    throw new Error("Cannot specify both fileIds and dataSources");
+  }
+  
+  return {
+    definition : { type: "code_interpreter" },
+    resources : { code_interpreter: { file_ids: fileIds, data_sources: dataSources } }
+  }
+}
+
+/**
+ * Creates an Azure AI search tool
+ * 
+ * @param indexConnectionId - The connection ID of the Azure AI search index.
+ * @param indexName - The name of the Azure AI search index.
+ * 
+ * @returns An object containing the definition and resources for the Azure AI search tool.
+ */
+export function createAzureAISearchTool( 
+  indexConnectionId: string,
+  indexName: string
+) : { definition : AzureAISearchToolDefinition, resources: ToolResources} {
+  return {
+    definition : { type: "azure_ai_search" },
+    resources : { azure_ai_search: { indexes: [{ index_connection_id: indexConnectionId, index_name: indexName  }] } }
+  }
+}
+
+/**
+ * Represents a set of tools with their definitions and resources.
+ */
+export class ToolSet {
+  /** A list of tool definitions that have been added to the tool set. */
+  toolDefinitions: ToolDefinition[] = [];
+
+  /** A collection of resources associated with the tools in the tool set. */
+  toolResources: ToolResources = {};
+
+  /**
+   * Adds a connection tool to the tool set.
+   * 
+   * @param toolType - The type of the connection tool.
+   * @param connectionIds - A list of the IDs of the connections to use.
+   */
+  addConnectionTool(
+    toolType: connectionToolType,
+    connectionIds: string[]
+  ) : void {
+    const tool = createConnectionTool(toolType, connectionIds);
+    this.toolDefinitions.push(tool.definition);
+  }
+
+  /**
+   * Adds a file search tool to the tool set.
+   * 
+   * @param vectorStoreIds - The ID of the vector store attached to this agent. There can be a maximum of 1 vector store attached to the agent.
+   * @param vectorStores -  The list of vector store configuration objects from Azure. This list is limited to one element. The only element of this list contains the list of azure asset IDs used by the search tool.
+   * @param definitionDetails - The input definition information for a file search tool as used to configure an agent.
+   */
+  addFileSearchTool(
+    vectorStoreIds?: string[],
+    vectorStores?: Array<VectorStoreConfigurations>,
+    definitionDetails?: FileSearchToolDefinitionDetails
+  ) : void {
+    const tool = createFileSearchTool(vectorStoreIds, vectorStores, definitionDetails);
+    this.toolDefinitions.push(tool.definition);
+    this.toolResources = { ...this.toolResources, ...tool.resources };
+  }
+
+  /**
+   * Adds a code interpreter tool to the tool set.
+   * 
+   * @param fileIds - A list of file IDs made available to the `code_interpreter` tool. There can be a maximum of 20 files associated with the tool.
+   * @param dataSources - The data sources to be used. This option is mutually exclusive with fileIds.
+   */
+  addCodeInterpreterTool(
```

Comment:
add and return tooldefintion and tool resource

Reply:
Can you clarify what you mean by this?

Reply:
This would add the tool and return the added tool / resource
`addCodeInterpreterTool(...):{toolDefintion:<>, toolResource:<>}`